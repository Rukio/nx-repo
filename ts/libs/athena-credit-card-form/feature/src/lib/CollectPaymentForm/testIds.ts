export const COLLECT_PAYMENT_FORM_TEST_IDS = {
  SUCCESS_ALERT: 'collect-payment-form-success-alert',
};
