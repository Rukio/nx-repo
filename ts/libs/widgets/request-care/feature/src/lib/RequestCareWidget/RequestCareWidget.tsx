import { FC, useState } from 'react';
import statsig from 'statsig-js';
import {
  RequestCareWidget as RequestCareForm,
  CareFor,
} from '@*company-data-covered*/widgets/request-care/ui';

export type RequestCareWidgetProps = {
  webRequestURL: string;
  marketingSiteURL: string;
};

export const getLearnMoreLink = (
  rootURL: string,
  relationshipToPatient: CareFor
) => {
  try {
    let path: string;
    switch (relationshipToPatient) {
      case CareFor.Myself:
        path = 'how-it-works';
        break;
      case CareFor.FamilyFriend:
        path = 'people-we-serve/caregivers';
        break;
      case CareFor.ClinicianOrganization:
      default:
        path = 'dispatchexpress';
        break;
    }

    return new URL(path, rootURL).href;
  } catch (error) {
    console.warn(`[Widgets app] getting learn more link failed:`, error);

    return '';
  }
};

export const getWebRequestLink = (
  rootURL: string,
  relationshipToPatient: CareFor
) => {
  try {
    return new URL(`?relationship=${relationshipToPatient}`, rootURL).href;
  } catch (error) {
    console.warn(`[Widgets app] getting web request link failed:`, error);

    return '';
  }
};

const RequestCareWidget: FC<RequestCareWidgetProps> = ({
  webRequestURL,
  marketingSiteURL,
}) => {
  const [relationshipToPatient, setRelationshipToPatient] = useState(
    CareFor.Myself
  );

  const isRequestCareEventToggleEnabled = statsig.checkGate(
    'requesting_care_for_page_traffic'
  );

  const onRelationshipToPatientChange = (value: CareFor) => {
    setRelationshipToPatient(value);
    if (isRequestCareEventToggleEnabled) {
      statsig.logEvent('requesting_care_for_dropdown_selection', value);
    }
  };

  const onWebRequestBtnClick = () => {
    if (isRequestCareEventToggleEnabled) {
      statsig.logEvent('main_page_schedule_appointment_button');
    }
  };

  const onLearnMoreBtnClick = () => {
    if (isRequestCareEventToggleEnabled) {
      statsig.logEvent('main_page_learn_more_button');
    }
  };

  const webRequestLink = getWebRequestLink(
    webRequestURL,
    relationshipToPatient
  );

  const learnMoreLink = getLearnMoreLink(
    marketingSiteURL,
    relationshipToPatient
  );

  return (
    <RequestCareForm
      onRelationshipToPatientChange={onRelationshipToPatientChange}
      webRequestLink={webRequestLink}
      learnMoreLink={learnMoreLink}
      onWebRequestBtnClick={onWebRequestBtnClick}
      onLearnMoreBtnClick={onLearnMoreBtnClick}
    />
  );
};

export default RequestCareWidget;
