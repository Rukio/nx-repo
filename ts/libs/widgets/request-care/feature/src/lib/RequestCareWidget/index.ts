export { default as RequestCareWidget } from './RequestCareWidget';
