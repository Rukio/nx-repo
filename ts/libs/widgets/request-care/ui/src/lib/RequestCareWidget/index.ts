export { default as RequestCareWidget } from './RequestCareWidget';
export { REQUEST_CARE_WIDGET_TEST_IDS } from './testIds';
