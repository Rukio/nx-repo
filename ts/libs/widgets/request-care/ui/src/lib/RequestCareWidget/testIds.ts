export const REQUEST_CARE_WIDGET_TEST_IDS = {
  CONTAINER: 'request-care-widget-container',
  RELATIONSHIP_SELECT: 'request-care-widget-relationship-select',
  RELATIONSHIP_SELECT_LABEL: 'request-care-widget-relationship-select-label',
  RELATIONSHIP_SELECT_OPTION: 'request-care-widget-relationship-select-option',
  REQUEST_A_VISIT_BUTTON: 'request-a-visit-button',
  LEARN_MORE_BUTTON: 'learn-more-button',
};
