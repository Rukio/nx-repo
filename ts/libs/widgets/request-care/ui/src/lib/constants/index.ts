export enum CareFor {
  Myself = 'patient',
  FamilyFriend = 'family:friend',
  ClinicianOrganization = 'clinician',
}
