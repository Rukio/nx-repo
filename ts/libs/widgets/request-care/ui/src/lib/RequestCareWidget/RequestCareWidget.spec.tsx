import {
  fireEvent,
  render,
  screen,
  waitFor,
  within,
} from '@testing-library/react';
import { CareFor } from '../constants';
import RequestCareWidget from './RequestCareWidget';
import { REQUEST_CARE_WIDGET_TEST_IDS } from './testIds';

const mockRequestCareWidgetProps = {
  onRelationshipToPatientChange: jest.fn(),
  webRequestLink: 'webRequestLink',
  learnMoreLink: 'learnMoreLink',
  onWebRequestBtnClick: jest.fn(),
  onLearnMoreBtnClick: jest.fn(),
};

describe('RequestCareWidget', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <RequestCareWidget {...mockRequestCareWidgetProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct relationship options', async () => {
    render(<RequestCareWidget {...mockRequestCareWidgetProps} />);

    const relationshipSelect = screen.getByTestId(
      REQUEST_CARE_WIDGET_TEST_IDS.RELATIONSHIP_SELECT
    );
    const relationshipSelectButton =
      within(relationshipSelect).getByRole('button');

    fireEvent.mouseDown(relationshipSelectButton);

    const relationshipMenu = await screen.findByRole('presentation');

    const relationshipOptions = within(
      relationshipMenu
    ).getAllByTestId<HTMLOptionElement>(
      REQUEST_CARE_WIDGET_TEST_IDS.RELATIONSHIP_SELECT_OPTION
    );

    expect(relationshipOptions.length).toBe(3);

    expect(relationshipOptions[0].selected).toBeTruthy();
    expect(relationshipOptions[0].textContent).toContain('Myself');

    expect(relationshipOptions[1].textContent).toContain(
      'A friend or family member'
    );

    expect(relationshipOptions[2].textContent).toContain(
      'As a clinician or organization'
    );
  });

  it('should change relationship correctly', async () => {
    render(<RequestCareWidget {...mockRequestCareWidgetProps} />);

    const relationshipSelect = screen.getByTestId(
      REQUEST_CARE_WIDGET_TEST_IDS.RELATIONSHIP_SELECT
    );
    const relationshipSelectButton =
      within(relationshipSelect).getByRole('button');

    fireEvent.mouseDown(relationshipSelectButton);

    const relationshipMenu = await screen.findByRole('presentation');

    const relationshipOptions = within(
      relationshipMenu
    ).getAllByTestId<HTMLOptionElement>(
      REQUEST_CARE_WIDGET_TEST_IDS.RELATIONSHIP_SELECT_OPTION
    );

    const newRelationshipOption = relationshipOptions[1];
    expect(newRelationshipOption.selected).toBeFalsy();

    fireEvent.click(newRelationshipOption);

    await waitFor(() =>
      expect(
        within(relationshipSelect).getByRole('button').textContent
      ).toContain('A friend or family member')
    );

    expect(
      mockRequestCareWidgetProps.onRelationshipToPatientChange
    ).toBeCalledWith(CareFor.FamilyFriend);
  });

  it('should call onWebRequestBtnClick if "Request a visit" button was clicked', async () => {
    render(<RequestCareWidget {...mockRequestCareWidgetProps} />);

    const requestAVisitButton = await screen.findByTestId(
      REQUEST_CARE_WIDGET_TEST_IDS.REQUEST_A_VISIT_BUTTON
    );

    fireEvent.click(requestAVisitButton);

    await waitFor(() => {
      expect(mockRequestCareWidgetProps.onWebRequestBtnClick).toBeCalled();
    });
  });

  it('should call onLearnMoreBtnClick if "Learn more" button was clicked', async () => {
    render(<RequestCareWidget {...mockRequestCareWidgetProps} />);

    const learnMoreButton = await screen.findByTestId(
      REQUEST_CARE_WIDGET_TEST_IDS.LEARN_MORE_BUTTON
    );

    fireEvent.click(learnMoreButton);

    await waitFor(() => {
      expect(mockRequestCareWidgetProps.onLearnMoreBtnClick).toBeCalled();
    });
  });
});
