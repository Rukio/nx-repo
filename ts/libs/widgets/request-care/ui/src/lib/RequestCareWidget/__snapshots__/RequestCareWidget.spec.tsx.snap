// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestCareWidget should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  z-index: 2;
  height: 100%;
  padding-top: 300px;
}

@media (min-width:900px) {
  .emotion-0 {
    padding-top: 400px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    background: #fff;
    padding-top: 0px;
    border-top-left-radius: 4px;
    border-bottom-left-radius: 4px;
    max-width: 641px;
    position: absolute;
    top: calc(50% - 340px);
    left: auto;
    right: 0;
    height: auto;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-bottom: 32px;
  width: 100%;
  background: #fff;
}

@media (min-width:600px) {
  .emotion-1 {
    padding-top: 32px;
    padding-bottom: 32px;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    padding-top: 40px;
    padding-bottom: 40px;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    background: none;
    padding: 80px 0px 100px;
  }
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 6rem;
  line-height: 1.167;
  letter-spacing: -0.01562em;
  color: #fff;
  position: relative;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  top: -210px;
  margin-bottom: -40px;
  font-size: 38px;
  line-height: 40px;
  padding-left: 20px;
  padding-right: 20px;
}

@media (min-width:600px) {
  .emotion-2 {
    padding-left: 40px;
    padding-right: 40px;
  }
}

@media (min-width:900px) {
  .emotion-2 {
    padding-left: 20px;
    padding-right: 20px;
    font-size: 60px;
    line-height: 64px;
    top: -310px;
    max-width: 600px;
    margin-bottom: -120px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    padding: 0px;
    color: #1976d2;
    text-align: start;
    padding-right: 97px;
    padding-left: 120px;
    top: 0;
    margin-bottom: 0px;
    margin-left: 0px;
    margin-right: 0px;
    max-width: initial;
  }
}

.emotion-3 {
  color: rgba(0, 0, 0, 0.87);
  padding-left: 16px;
  padding-right: 16px;
  max-width: 600px;
  margin-left: auto;
  margin-right: auto;
}

@media (min-width:900px) {
  .emotion-3 {
    padding-left: 20px;
    padding-right: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-3 {
    padding-left: 120px;
    padding-right: 120px;
    max-width: 100%;
  }
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  text-align: center;
  line-height: 33px;
}

@media (min-width:1200px) {
  .emotion-4 {
    text-align: start;
    margin-top: 16px;
    font-size: 20px;
    line-height: 33px;
  }
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  margin-top: 24px;
}

@media (min-width:1200px) {
  .emotion-5 {
    margin-top: 40px;
  }
}

.emotion-6 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
  margin-top: 8px;
}

.emotion-6.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-6:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-6:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-6.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-6.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-6.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-7 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 4px;
  cursor: pointer;
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
}

.emotion-7:focus {
  border-radius: 4px;
}

.emotion-7::-ms-expand {
  display: none;
}

.emotion-7.Mui-disabled {
  cursor: default;
}

.emotion-7[multiple] {
  height: auto;
}

.emotion-7:not([multiple]) option,
.emotion-7:not([multiple]) optgroup {
  background-color: #fff;
}

.emotion-7.emotion-7.emotion-7 {
  padding-right: 32px;
}

.emotion-7.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-7::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:focus {
  outline: 0;
}

.emotion-7:invalid {
  box-shadow: none;
}

.emotion-7::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-7.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-7:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-7:-webkit-autofill {
  border-radius: inherit;
}

.emotion-8 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.emotion-9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  position: absolute;
  right: 7px;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0, 0, 0, 0.54);
}

.emotion-9.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-10 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-11 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -16px;
  width: calc(100% + 16px);
  margin-left: -16px;
  padding-top: 16px;
}

.emotion-12>.MuiGrid-item {
  padding-top: 16px;
}

.emotion-12>.MuiGrid-item {
  padding-left: 16px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 55.833333%;
  -ms-flex-preferred-size: 55.833333%;
  flex-basis: 55.833333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 55.833333%;
}

@media (min-width:600px) {
  .emotion-13 {
    -webkit-flex-basis: 55.833333%;
    -ms-flex-preferred-size: 55.833333%;
    flex-basis: 55.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 55.833333%;
  }
}

@media (min-width:900px) {
  .emotion-13 {
    -webkit-flex-basis: 61.666667%;
    -ms-flex-preferred-size: 61.666667%;
    flex-basis: 61.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 61.666667%;
  }
}

@media (min-width:1200px) {
  .emotion-13 {
    -webkit-flex-basis: 61.666667%;
    -ms-flex-preferred-size: 61.666667%;
    flex-basis: 61.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 61.666667%;
  }
}

@media (min-width:1536px) {
  .emotion-13 {
    -webkit-flex-basis: 61.666667%;
    -ms-flex-preferred-size: 61.666667%;
    flex-basis: 61.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 61.666667%;
  }
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
}

.emotion-14::-moz-focus-inner {
  border-style: none;
}

.emotion-14.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-14 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-14:hover {
    background-color: #1976d2;
  }
}

.emotion-14:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-14.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-14.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-15 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 44.166667%;
  -ms-flex-preferred-size: 44.166667%;
  flex-basis: 44.166667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 44.166667%;
}

@media (min-width:600px) {
  .emotion-16 {
    -webkit-flex-basis: 44.166667%;
    -ms-flex-preferred-size: 44.166667%;
    flex-basis: 44.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 44.166667%;
  }
}

@media (min-width:900px) {
  .emotion-16 {
    -webkit-flex-basis: 38.333333%;
    -ms-flex-preferred-size: 38.333333%;
    flex-basis: 38.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 38.333333%;
  }
}

@media (min-width:1200px) {
  .emotion-16 {
    -webkit-flex-basis: 38.333333%;
    -ms-flex-preferred-size: 38.333333%;
    flex-basis: 38.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 38.333333%;
  }
}

@media (min-width:1536px) {
  .emotion-16 {
    -webkit-flex-basis: 38.333333%;
    -ms-flex-preferred-size: 38.333333%;
    flex-basis: 38.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 38.333333%;
  }
}

.emotion-17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(25, 118, 210, 0.5);
  color: #1976d2;
  width: 100%;
}

.emotion-17::-moz-focus-inner {
  border-style: none;
}

.emotion-17.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-17 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-17:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
}

@media (hover: none) {
  .emotion-17:hover {
    background-color: transparent;
  }
}

.emotion-17.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

<div
    class="MuiBox-root emotion-0"
    data-testid="request-care-widget-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item emotion-1"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 emotion-2"
      >
        We see patients differently
      </h1>
      <div
        class="MuiBox-root emotion-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-4"
        >
          In your home. On your terms. We bring the power of the hospital to you.
        </p>
        <h6
          class="MuiTypography-root MuiTypography-h6 emotion-5"
          data-testid="request-care-widget-relationship-select-label"
        >
          Request care for
        </h6>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  emotion-6"
          data-testid="request-care-widget-relationship-select"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input emotion-7"
            role="button"
            tabindex="0"
          >
            Myself
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput emotion-8"
            tabindex="-1"
            value="patient"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined emotion-9"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-10"
          >
            <legend
              class="emotion-11"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 emotion-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6.7 MuiGrid-grid-md-7.4 emotion-13"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth emotion-14"
              data-testid="request-a-visit-button"
              href="webRequestLink"
              tabindex="0"
            >
              Request a visit
              <span
                class="MuiTouchRipple-root emotion-15"
              />
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5.3 MuiGrid-grid-md-4.6 emotion-16"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth emotion-17"
              data-testid="learn-more-button"
              href="learnMoreLink"
              tabindex="0"
            >
              Learn More
              <span
                class="MuiTouchRipple-root emotion-15"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
